nexenta-fusion-installer.sh

```diff
@@ -0,0 +1,320 @@
+    else 
+        return 1
+    fi
+}
+
+# text functions 
+echoBlue() {
+    echo -e "${textBlue}${1}${textNc}"
+}
+
+echoRed() {
+    echo -e "${textRed}${1}${textNc}"
+}
+
+echoError() {
+    echo -e "${textRed}Error: ${1}${textNc}" >&2
+}
+
+echoDefaults() {
+    ask "Defaults have been selected for the following parameters:" "ESDB heap size is the memory reserved for the analytics database. The default recommendation is half of the total system memory."
+    echo "ESDB heap size:${textBold} ${defaultHeapSize} ${textNormal}"
+    echo "Timezone:${textBold} $OLSONTZ ${textNormal}"
+    echo "NexentaFusion folders path:${textBold} $defaultFusionPath ${textNormal}"
+}
+#
+
+runContainer() {
+    managementIp=${ips[$selectedIpIndex]}
+    heapSize=$typedHeapSize
+    
+    if [ -z "$typedTZ" ]; then
+        tz=$OLSONTZ
+    else
+        tz=$typedTZ
+    fi
+
+    if [ -z "$typedFusionPath" ]; then
+        path=$defaultFusionPath
+    else 
+        path=$typedFusionPath
+    fi
+
+    if [ -z "$typedHeapSize" ]; then
+        heapSize=$defaultHeapSize
+    else 
+        heapSize=$typedHeapSize
+    fi
+
+    # show summary if defaults were not accepted
+    if [ $1 = "n" ]; then
+        echoBlue "The NexentaFusion container will be run with the following parameters:"
+        echo "Management IP:${textBold} $managementIp${textNormal}"
+        echo "ESDB heap size:${textBold} ${heapSize} ${textNormal}"
+        echo "Timezone:${textBold} $tz ${textNormal}"
+        echo "NexentaFusion folders path:${textBold} $path ${textNormal}"
+        echo
+        echo "Press any key to continue"
+        read
+    fi
+
+    echoBlue "Running the NexentaFusion container..."
+    dockerRunCommand="sudo docker run --name $containerName -v $path/elasticsearch:/var/lib/elasticsearch -v $path/nef:/var/lib/nef -e MGMT_IP=$managementIp --ulimit nofile=65536:65536 --ulimit memlock=-1:-1 -e ES_HEAP_SIZE=$heapSize -e TZ=$tz -p 8457:8457 -p 9200:9200 -p 8443:8443 -i -d nexenta/fusion"
+
+    # check if image is pulled
+    if [ -z "$(sudo docker images -f "reference=nexenta/fusion:latest" | grep nexenta/fusion)" ]; then
+        $dockerRunCommand
+        else
+        # hide docker run output in case of existing image (we don't want to display a created container id)
+        $dockerRunCommand 1> /dev/null
+    fi
+
+    if [ $? -gt 0 ]; then
+        echoRed "Error during running docker container"
+        exit 1
+    fi
+
+    echo -e "Container with name ${textLightGray}$containerName${textNc} was created" 
+    
+    echoBlue "Waiting for NexentaFusion to start"
+    uiStatus="down"
+    while [ $uiStatus = "down" ]; do
+        echo -n .
+        uiStatus=$(getFusionUiStatus ${managementIp})
+        sleep 1
+    done
+    echo
+    echoBlue "NexentaFusion is available at https://${managementIp}:8457"
+}
+
+### WIZARD START
+
+# start script as sudo
+if (( $EUID != 0 )); then
+    echo "Starting nexenta-fusion-installer as sudo"
+    echo
+    sudo $0 "$@"
```

Comment:
Better use
```bash
exec sudo $0 "$@"
```
And no `exit` 

See [bash](http://manpages.ubuntu.com/manpages/trusty/en/man1/bash.1.html) description 

![image](https://user-images.githubusercontent.com/8813694/34236562-02a1986a-e5ad-11e7-8be0-d00618787e37.png)

Reply:
cool. Thanks!